src/commonMain/kotlin/fr/acinq/lightning/db/PaymentsDb.kt

```diff
@@ -127,7 +127,7 @@ data class IncomingPayment(val preimage: ByteVector32, val origin: Origin, val r
     override val completedAt: Long?
         get() = when {
             received == null -> null // payment has not yet been received
-            received.receivedWith.any { it is ReceivedWith.OnChainIncomingPayment && it.confirmedAt == null } -> null // payment has been received, but there is at least one unconfirmed on-chain part
+            received.receivedWith.any { it is ReceivedWith.OnChainIncomingPayment && it.lockedAt == null } -> null // payment has been received, but there is at least one unconfirmed on-chain part
```

Comment:
Shouldn't we still be looking at `confirmedAt` here?

Reply:
When defined, `completedAt` means that the received payment is usable, so it's indeed `lockedAt`.

Reply:
That field is also used for pure LN payments, where there is no concept of locking.

Reply:
But you agree it's very confusing for readers of that code (and most likely for wallets integrating it as well)? So we should do something, at least a comment to explain that confusion until we find a better way of expressing the distinction between `locked` / `confirmed` in terms of UX